flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java

```diff
@@ -473,6 +482,15 @@ private boolean shouldRoll(BucketState<T> bucketState) throws IOException {
 					subtaskIndex,
 					writePosition,
 					batchSize);
+			} else {
+				long currentProcessingTime = processingTimeService.getCurrentProcessingTime();
```

Comment:
You should pass in the current processing time to avoid calling this somewhat costly method twice.

Reply:
Updated method signature for `shouldRoll` to include the `currentProcessingTime`